apps/wallet-dashboard/components/Dialogs/Staking/views/DetailsView.tsx

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import {
+    ExtendedDelegatedStake,
+    ImageIcon,
+    ImageIconSize,
+    useFormatCoin,
+    formatPercentageDisplay,
+    useValidatorInfo,
+} from '@iota/core';
+import {
+    Header,
+    Button,
+    ButtonType,
+    Card,
+    CardBody,
+    CardImage,
+    CardType,
+    Panel,
+    KeyValueInfo,
+    Badge,
+    BadgeType,
+    Divider,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    LoadingIndicator,
+} from '@iota/apps-ui-kit';
+import { Warning } from '@iota/ui-icons';
+import { formatAddress, IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { Layout, LayoutFooter, LayoutBody } from './Layout';
+
+interface StakeDialogProps {
+    stakedDetails: ExtendedDelegatedStake;
+    showActiveStatus?: boolean;
+    handleClose: () => void;
+    handleUnstake: () => void;
+    handleStake: () => void;
+}
+
+export function DetailsView({
+    handleClose,
+    handleUnstake,
+    handleStake,
+    stakedDetails,
+    showActiveStatus,
+}: StakeDialogProps): JSX.Element {
+    const totalStake = BigInt(stakedDetails?.principal || 0n);
+    const validatorAddress = stakedDetails?.validatorAddress;
+    const {
+        isAtRisk,
+        isPendingValidators,
+        errorValidators,
+        validatorSummary,
+        apy,
+        isApyApproxZero,
+        newValidator,
+        commission,
+    } = useValidatorInfo({
+        validatorAddress,
+    });
+
+    const iotaEarned = BigInt(stakedDetails?.estimatedReward || 0n);
+    const [iotaEarnedFormatted, iotaEarnedSymbol] = useFormatCoin(iotaEarned, IOTA_TYPE_ARG);
+    const [totalStakeFormatted, totalStakeSymbol] = useFormatCoin(totalStake, IOTA_TYPE_ARG);
+
+    const validatorName = validatorSummary?.name || '';
+
+    const subtitle = showActiveStatus ? (
+        <div className="flex items-center gap-1">
+            {formatAddress(validatorAddress)}
+            {newValidator && <Badge label="New" type={BadgeType.PrimarySoft} />}
+            {isAtRisk && <Badge label="At Risk" type={BadgeType.PrimarySolid} />}
+        </div>
+    ) : (
+        formatAddress(validatorAddress)
+    );
+
+    if (isPendingValidators) {
+        return (
+            <div className="flex h-full w-full items-center justify-center p-2">
+                <LoadingIndicator />
+            </div>
+        );
+    }
+
+    if (errorValidators) {
+        return (
+            <div className="mb-2 flex h-full w-full items-center justify-center p-2">
+                <InfoBox
+                    title="Something went wrong"
+                    supportingText={'An error occurred'}
+                    style={InfoBoxStyle.Default}
+                    type={InfoBoxType.Error}
+                    icon={<Warning />}
+                />
+            </div>
+        );
+    }
```

Comment:
I think this is not necessary. We can show the stake information even if validator query fails, and set as `--` the information regarding the validator. wdyt?